File: Problem11/zigzagrec.py
from ast import List
from typing import Optional
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val=val
        self.left=left
        self.right=right
class Solution:
    ans=[]
    def zigzagrec(root,lev):
        if(root==None):
            return 
        if(len(Solution.ans)==lev):
            Solution.ans.append([])
        Solution.ans[lev].append(root.val)
        Solution.zigzagrec(root.left,lev+1)
        Solution.zigzagrec(root.right,lev+1)
    def zigzagLevelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        Solution.ans.clear()
        Solution.zigzagrec(root,0)
        for i in range(1,len(Solution.ans),2):
            Solution.ans[i].reverse()
        return Solution.ans

File: Problem11/zigzagrec.cpp
#include<bits/stdc++.h>
struct TreeNode
{
    int val;
    TreeNode *left,*right;
};

class Solution {
public:
    std::vector<std::vector<int>>ans;
    void level_orderrec(TreeNode*root,int lev)
    {
        if(root==NULL)
              return ;
        if(lev==ans.size())
               ans.push_back({});
        ans[lev].push_back(root->val);
        level_orderrec(root->left,lev+1);
        level_orderrec(root->right,lev+1);
    }
        
    std::vector<std::vector<int>> zigzagLevelOrder(TreeNode* root) {
        ans.clear();
        level_orderrec(root,0);
        for(int i=1;i<ans.size();i+=2)
             std::reverse(ans[i].begin(),ans[i].end());
        return ans;
        
    }
};